backend/lucene/src/main/java/org/hibernate/search/backend/lucene/impl/LuceneBackendFactory.java

```diff
@@ -160,6 +173,25 @@ private LuceneAnalysisDefinitionRegistry getAnalysisDefinitionRegistry(
 		}
 	}
 
+	private void configureQueryCashe(
+			BackendBuildContext buildContext, ConfigurationPropertySource propertySource,
+			QueryCachingConfigurationContext context) {
+		try {
+			LuceneCachingServiceConfigurer.configure( buildContext, context );
```

Comment:
I don't think we discussed this? You're essentially applying all caching configurers you can find, just because they are in the classpath, and without having to insert anything in the configuration properties.

Personally, I think this kind of "magic" brings more headaches than it helps and I don't think we should do it.

But... if you really, really want such a feature, please do it the Hibernate Search way:

```java
try ( BeanHolder<List<QueryCachingConfigurer>> implicitConfigurersHolder =
        beanResolver.resolve( beanResolver.allConfiguredForRole( QueryCachingConfigurer.class ) ) ) {
    for ( QueryCachingConfigurer configurer : implicitConfigurersHolder.get() ) {
        configurer.configure( context );
    }
}
```

Developers could then register their configurers this way, and they would be applied automatically:

```java
@MetaInfServices(BeanConfigurer.class)
public class PortalBeanConfigurer implements BeanConfigurer {
	@Override
	public void configure(BeanConfigurationContext context) {
		context.define(
				QueryCachingConfigurer.class,
				BeanReference.ofInstance((QueryCachingConfigurer) context -> {
					context.queryCache(new PortalQueryCache());
					context.queryCachingPolicy(new PortalQueryCachingPolicy());
				})
		);
	}
}
```

Reply:
I'll change it.